tvloader/parser2v3/parse_file.go

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+
+package parser2v3
+
+import (
+	"fmt"
+
+	"github.com/spdx/tools-golang/spdx/common"
+	"github.com/spdx/tools-golang/spdx/v2_3"
+)
+
+func (parser *tvParser2_3) parsePairFromFile2_3(tag string, value string) error {
+	// expire fileAOP for anything other than an AOPHomePage or AOPURI
+	// (we'll actually handle the HomePage and URI further below)
+	if tag != "ArtifactOfProjectHomePage" && tag != "ArtifactOfProjectURI" {
+		parser.fileAOP = nil
+	}
+
+	switch tag {
+	// tag for creating new file section
+	case "FileName":
+		// check if the previous file contained an spdx Id or not
+		if parser.file != nil && parser.file.FileSPDXIdentifier == nullSpdxElementId2_3 {
+			return fmt.Errorf("file with FileName %s does not have SPDX identifier", parser.file.FileName)
+		}
+		parser.file = &v2_3.File{}
+		parser.file.FileName = value
+	// tag for creating new package section and going back to parsing Package
+	case "PackageName":
+		parser.st = psPackage2_3
+		// check if the previous file contained an spdx Id or not
+		if parser.file != nil && parser.file.FileSPDXIdentifier == nullSpdxElementId2_3 {
+			return fmt.Errorf("file with FileName %s does not have SPDX identifier", parser.file.FileName)
+		}
+		parser.file = nil
+		return parser.parsePairFromPackage2_3(tag, value)
+	// tag for going on to snippet section
+	case "SnippetSPDXID":
+		parser.st = psSnippet2_3
+		return parser.parsePairFromSnippet2_3(tag, value)
+	// tag for going on to other license section
+	case "LicenseID":
+		parser.st = psOtherLicense2_3
+		return parser.parsePairFromOtherLicense2_3(tag, value)
+	// tags for file data
+	case "SPDXID":
+		eID, err := extractElementID(value)
+		if err != nil {
+			return err
+		}
+		parser.file.FileSPDXIdentifier = eID
+		if parser.pkg == nil {
+			if parser.doc.Files == nil {
+				parser.doc.Files = []*v2_3.File{}
+			}
+			parser.doc.Files = append(parser.doc.Files, parser.file)
+		} else {
+			if parser.pkg.Files == nil {
+				parser.pkg.Files = []*v2_3.File{}
+			}
+			parser.pkg.Files = append(parser.pkg.Files, parser.file)
+		}
+	case "FileType":
+		parser.file.FileTypes = append(parser.file.FileTypes, value)
+	case "FileChecksum":
+		subkey, subvalue, err := extractSubs(value)
+		if err != nil {
+			return err
+		}
+		if parser.file.Checksums == nil {
+			parser.file.Checksums = []common.Checksum{}
+		}
+		switch common.ChecksumAlgorithm(subkey) {
+		case common.SHA1, common.SHA256, common.MD5:
+			algorithm := common.ChecksumAlgorithm(subkey)
+			parser.file.Checksums = append(parser.file.Checksums, common.Checksum{Algorithm: algorithm, Value: subvalue})
```

Comment:
add case for newly defined unhandled checksums and create an issue as a TODO to add support for new checksums